server/src/main/scala/org/http4s/server/package.scala

```diff
@@ -43,4 +44,11 @@ package object server {
         .compose(AuthedRequest(authUser))
     }
   }
+
+  private[this] val messageFailureLogger = getLogger("org.http4s.server.message-failures")
+  def messageFailureHandler(req: Request): PartialFunction[Throwable, Task[Response]] = {
+    case mf: MessageFailure =>
+      messageFailureLogger.debug(mf)(s"Message failure handling request: ${req.method} ${req.pathInfo}")
```

Comment:
I think the remote IP is reasonable.  I've been wondering about a better toString/show for Request and Response.

I think customizing this would be tricky, other than shutting it off and adding a middleware that intercepts.  Proper request logging from blaze is a problem I'm hoping to tackle in that HttpEvent scheme I posted yesterday.  Those events are fit for both instruementation and logging coming out of blaze server and client.